yarn-project/types/src/abi/contract_artifact.ts

```diff
@@ -125,6 +135,19 @@ function generateFunctionArtifact(fn: NoirCompiledContractFunction): FunctionArt
   };
 }
 
+function getFunctionType(fn: NoirCompiledContractFunction): FunctionType {
+  if (fn.custom_attributes.includes(AZTEC_PRIVATE_ATTRIBUTE)) {
+    return FunctionType.SECRET;
+  } else if (fn.custom_attributes.includes(AZTEC_PUBLIC_ATTRIBUTE)) {
+    return FunctionType.OPEN;
+  } else if (fn.is_unconstrained) {
+    return FunctionType.UNCONSTRAINED;
+  } else {
+    // Default to a private function (see simple_macro_example_expanded for an example of this behavior)
+    return FunctionType.SECRET;
+  }
```

Comment:
One janky solution to this is just check to see if the function is a single opcode. Dealing with the private context ensures that no secret function can be a single opcode so this should be a sufficient check.